### PR TITLE
finish initial unf*ck of the frontend

### PR DESCRIPTION
**Pull request by dsifford**
The main goal of this PR was to get authentication and keypair association working for all login types (email/password, google, etc).

This PR does do that, but it also fixes a bunch of trash throughout the entire app.

**PLEASE** take some time and look this over and let me know if you have any questions.

### CODE REVIEW

**Review by ej3076 — APPROVED** [review]
🏅

File: frontend/package.json [hunk 1]
```diff
@@ -11,48 +11,49 @@
   },
```
Comment (ej3076) [inline frontend/package.json:11]: There are a lot of big dependency updates here - is this all ok / necessary?

File: frontend/src/app/auth/auth.service.ts [hunk 2]
```diff
@@ -1,140 +1,162 @@
...
+        this.errorMessage = e.message;
+        console.error(this.errorMessage);
+      });
   }
 
-  setManageApi(accessToken: string): void {
-    this.auth0Manage = new auth0.Management({
-      domain: AUTH_CONFIG.CLIENT_DOMAIN,
-      token: accessToken
+  private async setUserKeys(accessToken: string, id: string): Promise<User> {
```
Comment (ej3076) [inline frontend/src/app/auth/auth.service.ts:147]: Nice.

File: frontend/src/app/edit-claim/edit-claim.component.ts [hunk 3]
```diff
@@ -1,62 +1,64 @@
...
+    if (event.currentTarget instanceof HTMLInputElement) {
+      this.files = event.currentTarget.files;
+    }
   }
 
   deleteFile(hash: string): void {
-    this.service.deleteFile(hash, this.claim.vehicle.vin).subscribe();
-
-    //TODO: show visual confirmation of deletion
+    // FIXME: This is not correct
```
Comment (ej3076) [inline frontend/src/app/edit-claim/edit-claim.component.ts:51]: This will be paired with downloads 🌡

File: frontend/src/app/auth/auth.service.ts [hunk 4]
```diff
@@ -1,140 +1,162 @@
...
-import { BehaviorSubject } from 'rxjs';
-import { AUTH_CONFIG } from './auth.config';
-import * as auth0 from 'auth0-js';
-import { ENV } from '../core/env.config';
+import {
+  Auth0DecodedHash as DecodedHash,
+  Auth0UserProfile as User,
+  Management,
+  WebAuth,
+} from 'auth0-js';
```
Comment (sloan-seamus) [inline frontend/src/app/auth/auth.service.ts:9]: I like this a lot

File: frontend/src/app/auth/auth.service.ts (hunk 2 above)
Comment (sloan-seamus) [reply frontend/src/app/auth/auth.service.ts:147]: Nice.

File: frontend/src/app/new-claim/new-claim.component.ts [hunk 5]
```diff
@@ -66,12 +65,12 @@ export class NewClaimComponent implements OnInit {
   }
 
   public alphaNumericValidator(event: any) {
-    const charactersAllowed = /^[a-zA-Z0-9]*$/;   
+    const charactersAllowed = /^[a-zA-Z0-9]*$/;
```
Comment (sloan-seamus) [inline frontend/src/app/new-claim/new-claim.component.ts:68]: Was this just prettier or did you actually find this extra space

**Review by sloan-seamus — APPROVED** [review]

File: frontend/src/app/new-claim/new-claim.component.ts (hunk 5 above)
Comment (dsifford) [reply frontend/src/app/new-claim/new-claim.component.ts:68]: You'll never know :smiling_imp:

File: frontend/src/app/auth/auth.service.ts (hunk 4 above)
Comment (dsifford) [reply frontend/src/app/auth/auth.service.ts:9]: Not sure if sarcastic........ :thinking:

File: frontend/src/app/auth/auth.service.ts (hunk 2 above)
Comment (dsifford) [reply frontend/src/app/auth/auth.service.ts:147]: Nice.

**Pull request merged**